src/main/java/com/gd/clinic/security/service/RefreshTokenService.java

```diff
@@ -0,0 +1,57 @@
+package com.gd.clinic.security.service;
+
+import com.gd.clinic.security.repository.RefreshTokenRepository;
+import com.gd.clinic.security.entity.RefreshToken;
+import com.gd.clinic.security.entity.User;
+import com.gd.clinic.model.JwtRefreshRequestDto;
+import com.gd.clinic.model.JwtResponseDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class RefreshTokenService {
+
+    private final RefreshTokenRepository refreshTokenRepository;
+
+    @Value("${jwt.refreshToken.expiration}")
+    private int expiration;
+
+    public String createToken(User user) {
+        var refreshToken = refreshTokenRepository.save(RefreshToken.builder()
+                .token(UUID.randomUUID().toString())
+                .user(user)
+                .expiration(ZonedDateTime.now(ZoneId.systemDefault()).plusMinutes(expiration))
+                .build());
+        return refreshToken.getToken();
+    }
+
+    public JwtResponseDto refreshToken(JwtRefreshRequestDto refreshRequestDto) {
+        var tokenOpt = refreshTokenRepository.findRefreshTokenByToken(refreshRequestDto.getRefreshToken());
+        if (tokenOpt.isEmpty()) {
+            throw new RuntimeException("Refresh Token %s not found!".formatted(refreshRequestDto.getRefreshToken()));
+        }
+        var token = tokenOpt.get();
+        if (isTokenExpired(token.getExpiration())) {
+            refreshTokenRepository.delete(token);
+            throw new RuntimeException("Refresh Token %s was expired!".formatted(refreshRequestDto.getRefreshToken()));
+        }
+        String jwt = createToken(token.getUser());
+        updateToken(token);
+        return new JwtResponseDto(jwt, token.getToken());
+    }
+
+    private void updateToken(RefreshToken token) {
+        token.setExpiration(ZonedDateTime.now(ZoneId.systemDefault()).plusMinutes(expiration));
+        refreshTokenRepository.save(token);
+    }
+
+    private boolean isTokenExpired(ZonedDateTime expirationTime) {
+        return expirationTime.isBefore(ZonedDateTime.now(ZoneId.systemDefault()));
+    }
+}
```

Comment:
seems you copied it from somewhere because I haven't seen that any of you used `var` keyword. 
it's fine and actually good example of neat code